src/example/prompt.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class PullRequestReviewService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async getPullRequestDetails(repo: string, prId: number): Promise<any> {
+    const url = `https://api.github.com/repos/${repo}/pulls/${prId}`;
+    return this.httpService.get(url).pipe(map(response => response.data)).toPromise();
+  }
+
+  async reviewPullRequest(repo: string, prId: number): Promise<void> {
+    const details = await this.getPullRequestDetails(repo, prId);
+
+    if (details.changed_files > 10) {
+      this.addComment(repo, prId, "Too many files changed in this PR.");
+    }
+
+    if (!details.title.includes("JIRA-")) {
+      this.addComment(repo, prId, "PR title does not include JIRA ticket.");
+    }
+
+    this.mergePullRequest(repo, prId);
+  }
+
+  async addComment(repo: string, prId: number, comment: string): Promise<void> {
+    const url = `https://api.github.com/repos/${repo}/pulls/${prId}/comments`;
+    await this.httpService.post(url, { body: comment });
```

Comment:
PullProofAI

The `addComment` method should handle potential errors from the HTTP POST request to avoid unhandled promise rejections.
```diff
-     await this.httpService.post(url, { body: comment });
+     try {
+       await this.httpService.post(url, { body: comment });
+     } catch (error) {
+       throw new Error(`Failed to add comment: ${error.message}`);
+     }
```


<!-- This is an auto-generated comment by PullProofAI -->